mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraMessageIdMapper.java

```diff
@@ -130,38 +121,26 @@ public CassandraMessageIdMapper(MailboxMapper mailboxMapper, CassandraMailboxDAO
                         .completableFuture());
     }
 
-    private CompletableFuture<Boolean> mailboxExists(MessageWithoutAttachment messageWithoutAttachment) {
-        CassandraId cassandraId = (CassandraId) messageWithoutAttachment.getMailboxId();
+    private CompletableFuture<Stream<SimpleMailboxMessage>> filterMessagesWIthExistingMailbox(Stream<SimpleMailboxMessage> stream) {
+        return FluentFutureStream.of(stream.map(this::mailboxExists))
+            .flatMap(m -> m)
+            .completableFuture();
+    }
+
+    private CompletableFuture<Stream<SimpleMailboxMessage>> mailboxExists(SimpleMailboxMessage message) {
```

Comment:
Steam is more convenient than optionals, as you can flatMap streams, not optionals.

It's a Stream programming trick...